packages/treecrdt-core/tests/defensive_delete.rs

```diff
@@ -0,0 +1,672 @@
+use treecrdt_core::{LamportClock, MemoryStorage, NodeId, Operation, ReplicaId, TreeCrdt};
+
+#[test]
+fn delete_parent_then_insert_child_unaware() {
+    let mut crdt_a = TreeCrdt::new(
+        ReplicaId::new(b"a"),
+        MemoryStorage::default(),
+        LamportClock::default(),
+    );
+    let mut crdt_b = TreeCrdt::new(
+        ReplicaId::new(b"b"),
+        MemoryStorage::default(),
+        LamportClock::default(),
+    );
+
+    let root = NodeId::ROOT;
+    let parent = NodeId(1);
+    let child = NodeId(2);
+
+    // Both clients start with the same tree: parent under root
+    // Use local_insert to ensure proper setup
+    let insert_parent_op = crdt_a.local_insert(root, parent, 0).unwrap();
+    crdt_b.apply_remote(insert_parent_op).unwrap();
+
+    // Client A deletes the parent
+    let delete_op = crdt_a.local_delete(parent).unwrap();
+    assert!(crdt_a.is_tombstoned(parent));
+    assert!(crdt_a.children(parent).unwrap().is_empty());
+
+    // Client B doesn't know about deletion yet, inserts child under parent
+    // We need to ensure Client B's clock is advanced so the insert has higher lamport
+    // First, let Client B observe the delete's lamport (but don't apply it yet)
+    // Actually, we'll create the insert with a manually set higher lamport
+    let insert_child = Operation::insert(
+        &ReplicaId::new(b"b"),
+        1,
+        delete_op.meta.lamport + 1,
+        parent,
+        child,
+        0,
+    );
+    crdt_b.apply_remote(insert_child.clone()).unwrap();
+    assert_eq!(crdt_b.parent(child), Some(parent));
+    assert!(!crdt_b.is_tombstoned(parent)); // Parent not tombstoned in B's view
+
+    // Now synchronize: Client B receives the delete
+    crdt_b.apply_remote(delete_op.clone()).unwrap();
```

Comment:
I wonder actually here in a defensivie delted behaviour this should be 


```rs
// applying the remote deletion node will first apply deletion according to the kleppman spec, but this would also construct revival move

// pseudo code

let revival_ops = crdt_b.apply_remote(delete_op.clone(), { deleteBehaviour: 'defensive' }).unwrap();


crdt_a.apply_remote(insert_child).unwrap(); // no revival yet
crdt_a.apply_remote(revival_ops).unwrap(); // here we realize that client b reallise want that the child should exist!

```